application/src/main/kotlin/org/gxf/soapbridge/valueobjects/ProxyServerRequestMessage.kt

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: Copyright Contributors to the GXF project
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.gxf.soapbridge.valueobjects
+
+import mu.KotlinLogging
+import org.gxf.soapbridge.exceptions.ProxyMessageException
+import java.util.*
+
+
+class ProxyServerRequestMessage(
+    connectionId: String,
+    val commonName: String,
+    val context: String,
+    val soapPayload: String
+) : ProxyServerBaseMessage(connectionId) {
+
+    override fun getFieldsForMessage(): List<String> = listOf(
+        connectionId,
+        encode(context),
+        encode(soapPayload),
+        encode(commonName)
+    )
+
+    companion object {
+        private val LOGGER = KotlinLogging.logger { }
+
+        /**
+         * Constructs a ProxyServerRequestMessage instance from a string separated by '~'.
+         *
+         * @param string The input string.
+         * @return A ProxyServerRequestMessage instance.
+         * @throws ProxyServerException
+         */
+        @Throws(ProxyMessageException::class)
+        fun createInstanceFromString(string: String): ProxyServerRequestMessage {
+            val split = string.split(SEPARATOR)
+            "een of andere string".lines()
+            val numTokens = split.size
+            LOGGER.debug("split.length: {}", numTokens)
+            if (numTokens < 4 || numTokens > 5) {
+                throw ProxyMessageException(
+                    "Invalid number of tokens, not trying to create ProxyServerRequestMessage."
+                )
+            }
+            if (LOGGER.isDebugEnabled) {
+                printValues(numTokens, split)
+            }
+            val connectionId = split[0]
+            val context = decode(split[1])
+            val soapRequest = decode(split[2])
+            val commonName: String
+            val signature: String
+            if (numTokens == 4) {
+                // No common name used.
+                commonName = ""
+                signature = split[3]
+            } else {
+                // Common name used.
+                commonName = decode(split[3])
+                signature = split[4]
+            }
+            val proxyServerRequestMessage =
+                ProxyServerRequestMessage(connectionId, commonName, context, soapRequest)
+            proxyServerRequestMessage.signature = signature
+            return proxyServerRequestMessage
+        }
+
+        private fun printValues(numTokens: Int, split: List<String?>) {
```

Comment:
Ik heb dit liever als toString op de `ProxyServerRequestMessage` in plaats van een print functie op het companion object.